contracts/actions/liquity/helpers/CBHelper.sol

```diff
@@ -0,0 +1,140 @@
+    function getOptimalLusdAmount(uint256 _lusdAmount) public view returns (uint256, uint256) {
+        CBInfo memory systemInfo = getCbInfo();
+        uint256 marketPrice = getBLusdPriceFromCurve(_lusdAmount);
+        uint256 optimalRebondTime = _getOptimalRebondTime(systemInfo, marketPrice);
+
+        uint256 feeAmount = marketPrice * systemInfo.chickenInAMMFee;
+        uint256 marketPriceMinusFee = (marketPrice * 10**18) - feeAmount;
+
+        uint256 res = wmul(
+            wdiv(
+                wmul(_lusdAmount, optimalRebondTime),
+                (systemInfo.accrualParameter + optimalRebondTime)
+            ),
+            marketPriceMinusFee
+        );
+
+        return (res / 1e18, marketPrice);
+    }
+
+    /// @notice Internal function calculated optimal wait time for the user to accrue bLUSD
+    function _getOptimalRebondTime(CBInfo memory systemInfo, uint256 _marketPrice)
+        internal
+        pure
+        returns (uint256)
+    {
+        uint256 marketPricePremium = _calcMarketPricePremium(systemInfo, _marketPrice);
+
+        uint256 feeAmount = systemInfo.chickenInAMMFee * marketPricePremium;
+        uint256 premiumMinusFee = (marketPricePremium * 1e18) - feeAmount;
+
+        uint256 premiumSqrt = premiumMinusFee.sqrt();
+        uint256 premiumScaled = (premiumMinusFee / 1e18);
+
+        uint256 res = wmul(
+            systemInfo.accrualParameter,
+            wmul((premiumSqrt + 1e18), wdiv(1e18, (premiumScaled - 1e18)))
+        );
+
+        return res;
+    }
+
+    /// @notice Calculates market price premium based on the floor price and the current market price
+    function _calcMarketPricePremium(CBInfo memory systemInfo, uint256 _marketPrice)
+        public
+        pure
+        returns (uint256 marketPricePremium)
+    {
+        uint256 floorPrice = wdiv(systemInfo.totalReserveLUSD, systemInfo.bLUSDSupply);
+        marketPricePremium = wdiv(_marketPrice, floorPrice);
+    }
+
+    /// @notice View helper for calculating rebond time without input params
+    function getOptimalRebondTime() public view returns (uint256) {
+        CBInfo memory systemInfo = getCbInfo();
+        uint256 marketPrice = getBLusdPriceFromCurve(1000 * 1e18);
+
+        return _getOptimalRebondTime(systemInfo, marketPrice);
+    }
+
+    /// @notice Returns info about cb system needed for the calculations
+    function getCbInfo() public view returns (CBInfo memory systemInfo) {
+        (, uint256 totalReserveLUSD, ) = CBManager.getTreasury();
+
+        systemInfo = CBInfo({
+            totalReserveLUSD: totalReserveLUSD,
+            accrualParameter: CBManager.calcUpdatedAccrualParameter(),
+            chickenInAMMFee: CBManager.CHICKEN_IN_AMM_FEE(),
+            bLUSDSupply: IERC20(BLUSD_ADDRESS).totalSupply()
+        });
+    }
+
+    function formatRebondSub(uint64 _strategyId, uint256 _newSubId, uint256 _bondID) public pure returns (StrategyModel.StrategySub memory rebondSub) {
```

Comment:
strategyId is in both cases the already hardcoded uint64 public constant REBOND_STRATEGY_ID = 31 that's in this contract, maybe remove it as a param in this function and just use that value inside it.